tailscale/config.yaml

```diff
@@ -41,3 +41,7 @@ schema:
   tags: ["match(^tag:[a-zA-Z0-9]-?[a-zA-Z0-9]+$)?"]
   taildrop: bool?
   userspace_networking: bool?
+ports:
+  443/tcp: 443
+ports_description:
+  443/tcp: Tailscale Proxy and Funnel port
```

Comment:
The addon is on the host network, so does it make sense to use the port mapping feature?, as it looks like this is then referenced in the scripts, would it make more sense to use a config option (where you could also define the list of valid options) ?

Reply:
It's true, but it was the suggested way in similar situation (https://github.com/hassio-addons/addon-tailscale/pull/170#discussion_r1157057299) to configure ports even when we are on the host network (most users won't know the diference).

Reply:
And eg. the SSH add-on works this way also, I've stolen the idea from there. https://github.com/hassio-addons/addon-ssh/blob/main/ssh/rootfs/etc/s6-overlay/s6-rc.d/init-ssh/run

Reply:
No worries, I get it, just as the requirement is ideally to have a selected port definition and in my mind ideally a default that isn't 443, it would make more sense.

However I will of course defer to Frenck :)

Reply:
I don't feel like this is used similar to what SSH does. While this does provide an interface, this is used the other way around. 

Not sure on this tbh 🤔

Reply:
Just from a user perspective if we need to restrict the option set, we can really only do that from a list anyway, and as everything is now in a nice GUI, does it make sense to use the Port Mapping inputs?

Reply:
I think in case of reverse proxy functionality, it works similar to SSH or NGINX or MQTT, this is a service published by the add-on.

Funnel is a tricky concept, yes. But Funnel can serve only the same ports (on the same URLs) that the reverse proxy publishes, so it is "only" publishes this service not only on the tailnet, but on the internet. So we can say, this is the same service, with wider network accessibility (like opening a port on the router won't change our concept of NGINX proxy functionality on port X).

Reply:
I don't think that works, as one basically maps X:Y. In case of the UI offered by Home Assistant, the user can't configure X, only Y.

Reply:
> I don't think that works, as one basically maps X:Y. In the case of the UI offered by Home Assistant, the user can't configure X, only Y. (And even that is in a limited way).

The current hard wired 443 value can collide with already existing services.

Reply:
Apologies, I'm not that familiar with Tailscale, but I think I get it now.  The proxy can be configured independently, and provides a reverse proxy on the tailnet which can run on any configured port.  I'm assuming that the port isn't availiable outside of the tailnet? (i.e. on the host)

Funnel is then a configured item that exposes via Tailscale's services (hence the port requirement), which then advertises via the Internet.

If they are independent, why are they referencing the same config item?

I would use something along the lines of:

Proxy - Port definition (although I think it's not really exposing a port)

Funnel - List of Nope, 443, 8443, 10000

Reply:
Yes, proxy can run on any port inside the tailnet.

> If they are independent, why are they referencing the same config item?

No, they are not independent.

Funnel can "choose" 3 ports (443, 8443, 10000) to publish on the internet. But those 3 ports have to be already existing Proxy ports, so "funneling" is only a "yes/no" decision on the user's side, he can't say, to run Proxy on port 9876 and publish it on the internet as 8443, the 2 ports has to be the same.